core/src/main/java/com/microsoft/applicationinsights/internal/channel/samplingV2/SamplingScoreGeneratorV2.java

```diff
@@ -0,0 +1,57 @@
+package com.microsoft.applicationinsights.internal.channel.samplingV2;
+
+import com.microsoft.applicationinsights.agent.internal.common.StringUtils;
+import com.microsoft.applicationinsights.telemetry.Telemetry;
+
+import java.util.Random;
+
+/**
+ * Created by Dhaval Doshi Oct 2017
+ * This class generates the sample using the random number generator.
+ * It also contains the logic to preserve the correlated telemetry items.
+ */
+public class SamplingScoreGeneratorV2 {
+
+    private static Random random = new Random();
```

Comment:
Is there potential for multi-threaded contention here? Random is thread-safe, ThreadLocalRandom performs better in multi-threaded environment. 
https://docs.oracle.com/javase/8/docs/api/java/util/Random.html?is-external=true

Reply:
@grlima yes I can have a look. BTW does this ThreadLocalRandom exist in Java 6 too? Documentation seems to be of Java 8 where as we are currently coding against JDK 1.6